plugins/greatexpectations/flytekitplugins/greatexpectations/schema.py

```diff
@@ -0,0 +1,294 @@
+import datetime
+import logging
+import os
+import typing
+from dataclasses import dataclass
+from typing import Type
+
+import great_expectations as ge
+from dataclasses_json import dataclass_json
+from great_expectations.checkpoint import SimpleCheckpoint
+from great_expectations.core.run_identifier import RunIdentifier
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.exceptions import ValidationError
+
+from flytekit import FlyteContext
+from flytekit.extend import TypeEngine, TypeTransformer
+from flytekit.models import types as _type_models
+from flytekit.models.literals import Literal, Primitive, Scalar
+from flytekit.models.types import LiteralType
+from flytekit.types.file.file import FlyteFile, FlyteFilePathTransformer
+from flytekit.types.schema.types import FlyteSchema, FlyteSchemaTransformer, SchemaOpenMode
+
+
+@dataclass_json
+@dataclass
+class BatchConfig(object):
+    """
+    Use this configuration to configure Batch Request. A BatchRequest can either be
+    a simple BatchRequest or a RuntimeBatchRequest.
+
+    Args:
+        data_connector_query: query to request data batch
+        runtime_parameters: parameters to be passed at runtime
+        batch_identifiers: identifiers to identify the data batch
+        batch_spec_passthrough: reader method if your file doesn't have an extension
+        limit: number of data points to fetch
+    """
+
+    data_connector_query: typing.Optional[typing.Dict[str, typing.Any]] = None
+    runtime_parameters: typing.Optional[typing.Dict[str, str]] = None
+    batch_identifiers: typing.Optional[typing.Dict[str, str]] = None
+    batch_spec_passthrough: typing.Optional[typing.Dict[str, typing.Any]] = None
+    limit: typing.Optional[int] = None
```

Comment:
1. `limit`: Ah! You're right. I was of the opinion that a user might want to use `limit` without a `data_connector_query`; used it henceforth. But I don't think there's a need to, hence modified it now!
2. I think what you're referring to is this code snippet:
```python
def test_ge_schema_batchrequest_pandas_config():
    @task
    def my_task(
        directory: GEType[
            str,
            GEConfig(
                data_source="data",
                expectation_suite="test.demo",
                data_connector="my_data_connector",
                batchrequest_config=BatchConfig(
                    data_connector_query={
                        "batch_filter_parameters": {
                            "year": "2019",
                            "month": "01",  # noqa: F722
                        },
                        "limit": 10,
                    },
                ),
            ),
        ]
    ) -> str:
        return directory

    @workflow
    def my_wf():
        my_task(directory="my_assets")

    my_wf()
```
This is present in `test_schema.py` file. Will this work?

Reply:
This is one of the ways to use it, but it's possible to pass in a `custom_filter_function` to `data_connector_query` instead of keyworded `batch_filter_parameters` as on line 454 in the test shared above. I don't know that this is strictly necessary to include here, but wanted to mention it in case you thought it might be useful.

Reply:
@talagluck: I tried running this test:
```python
def test_ge_batchrequest_custom_filter():
    def my_custom_batch_selector(batch_identifiers: dict) -> bool:
        return int(batch_identifiers["year"]) == "2019" and int(batch_identifiers["month"]) == "01"

    task_object = GreatExpectationsTask(
        name="test19",
        data_source="data",
        inputs=kwtypes(data=str),
        expectation_suite="test.demo",
        data_connector="my_data_connector",
        task_config=BatchRequestConfig(
            data_connector_query={
                "custom_filter_function": my_custom_batch_selector,
            },
        ),
    )

    task_object(data="my_assets")
```
which resulted in an error: 

```
../../../flytekit/core/base_task.py:298: in __call__
    return self.execute(**kwargs)
../../flytekit-greatexpectations/flytekitplugins/great_expectations/task.py:209: in execute
    checkpoint = SimpleCheckpoint(f"_tmp_checkpoint_{self._expectation_suite}", context, **checkpoint_config)
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/checkpoint/checkpoint.py:664: in __init__
    checkpoint_config: CheckpointConfig = self._configurator_class(
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/checkpoint/configurator.py:114: in build
    return self._build_checkpoint_config()
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/checkpoint/configurator.py:142: in _build_checkpoint_config
    f"SimpleCheckpointConfigurator built this CheckpointConfig:"
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/data_context/types/base.py:116: in to_json_dict
    return convert_to_json_serializable(data=commented_map)
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/core/util.py:144: in convert_to_json_serializable
    new_dict[str(key)] = convert_to_json_serializable(data[key])
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/core/util.py:151: in convert_to_json_serializable
    new_list.append(convert_to_json_serializable(val))
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/core/util.py:144: in convert_to_json_serializable
    new_dict[str(key)] = convert_to_json_serializable(data[key])
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/core/util.py:144: in convert_to_json_serializable
    new_dict[str(key)] = convert_to_json_serializable(data[key])
../../../../../.pyenv/versions/3.8.10/envs/ge/lib/python3.8/site-packages/great_expectations/core/util.py:144: in convert_to_json_serializable
    new_dict[str(key)] = convert_to_json_serializable(data[key])
TypeError: <function test_ge_batchrequest_custom_filter.<locals>.my_custom_batch_selector at 0x108ae9280> is of type function which cannot be serialized.
```
Is `custom_filter_function` not supported within `SimpleCheckpoint` class? Moreover, am I using `custom_filter_function` the right way?

Reply:
Hi @samhita-alla, it does appear that you are technically using `custom_filter_function` in the right way, though I suspect that you are right, and `custom_filter_functions` are not supported within `SimpleCheckpoint`.  I'm going to dig a little deeper and open an issue with the team, but in the meantime, I think it's fair to not use this example for right now.

Reply:
Thank you for the confirmation, @talagluck. We can add the test once the issue is fixed.